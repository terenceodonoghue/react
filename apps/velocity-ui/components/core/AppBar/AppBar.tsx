import React, { FunctionComponent, ReactNode } from 'react';

interface AppBarProps {
  children: ReactNode;
}

const AppBar: FunctionComponent<AppBarProps> = (props) => (
  <div
    css={({ palette, zIndex }) => ({
      alignItems: 'center',
      backgroundColor: palette.neutral[50],
      borderBottom: `solid 1px ${palette.neutral[300]}`,
      display: 'flex',
      minHeight: 80,
      padding: '0 25px',
      position: 'sticky',
      top: 0,
      zIndex: zIndex.appBar,
    })}
    data-testid="app-bar"
    {...props}
  />
);

export default AppBar;
