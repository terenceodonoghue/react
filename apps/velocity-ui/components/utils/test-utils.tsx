/* eslint-disable import/no-extraneous-dependencies */
import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider, {
  ThemeProviderProps,
} from '~/components/core/ThemeProvider';

const Provider: FunctionComponent<ThemeProviderProps> = ({ children }) => (
  <ThemeProvider>{children}</ThemeProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Provider, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
