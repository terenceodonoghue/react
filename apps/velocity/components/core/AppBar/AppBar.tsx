import { FunctionComponent, ReactNode } from 'react';

interface AppBarProps {
  children: ReactNode;
}

const AppBar: FunctionComponent<AppBarProps> = (props) => (
  <div
    css={({ color }) => ({
      alignItems: 'center',
      backgroundColor: color.neutral[50],
      borderBottom: `solid 1px ${color.neutral[300]}`,
      display: 'flex',
      minHeight: 80,
      padding: '0 25px',
      position: 'sticky',
      top: 0,
      zIndex: 1100,
    })}
    data-testid="app-bar"
    {...props}
  />
);

export default AppBar;
