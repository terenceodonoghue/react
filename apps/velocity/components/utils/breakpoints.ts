export const TABLET = 768;
export const DESKTOP = 1025;

const breakpoints = [TABLET, DESKTOP];

export default breakpoints;
