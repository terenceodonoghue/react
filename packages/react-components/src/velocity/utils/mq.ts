import facepaint from 'facepaint';

export type MediaQuery<Values> = Values | Partial<[Values, Values, Values]>;

const mq = facepaint(
  [768, 1025].map((bp) => `@media (min-width: ${bp}px)`),
  { overlap: true },
);

export default mq;
