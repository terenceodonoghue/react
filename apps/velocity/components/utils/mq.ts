import facepaint from 'facepaint';

import breakpoints from './breakpoints';

const mq = facepaint(
  breakpoints.map((bp) => `@media (min-width: ${bp}px)`),
  { overlap: true },
);

export default mq;
