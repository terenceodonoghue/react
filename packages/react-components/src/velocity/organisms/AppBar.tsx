/** @jsxImportSource @emotion/react */
import { useTheme } from '@emotion/react';
import type { FunctionComponent, HTMLAttributes } from 'react';

import { ReactIcon } from '@terenceodonoghue/react-icons';
import {
  Mail,
  Menu,
  Notifications,
} from '@terenceodonoghue/react-icons/velocity';

import Avatar from '../atoms/Avatar.js';
import IconButton from '../atoms/IconButton.js';
import Indicator from '../atoms/Indicator.js';
import VelocityLogo from '../atoms/Logo.js';
import mq from '../utils/mq.js';

export interface AppBarProps extends HTMLAttributes<HTMLDivElement> {
  avatar: string;
  hasMessage?: boolean;
  hasNotification?: boolean;
  logo?: ReactIcon;
  drawerHandler?: () => void;
  messageHandler?: () => void;
  notificationHandler?: () => void;
  profileHandler?: () => void;
}

const AppBar: FunctionComponent<AppBarProps> = ({
  avatar,
  hasMessage = false,
  hasNotification = false,
  logo: Logo = VelocityLogo,
  drawerHandler,
  messageHandler,
  notificationHandler,
  profileHandler,
  ...props
}) => {
  const { color, page } = useTheme();

  return (
    <header
      css={{
        position: 'sticky',
        top: 0,
        display: 'flex',
        alignItems: 'center',
        borderBottom: `solid 1px ${color.neutral[300]}`,
        minHeight: 80,
        backgroundColor: color.neutral[50],
      }}
      {...props}
    >
      {drawerHandler ? (
        <div css={{ padding: '0 24px', width: 80 }}>
          <IconButton aria-label="menu" onClick={drawerHandler}>
            <Menu color={color.neutral[400]} size={18} />
          </IconButton>
        </div>
      ) : undefined}
      <div
        css={mq({
          flex: 1,
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'space-between',
          maxWidth: page.maxWidth,
          padding: ['0 24px', undefined, '0 110px'],
        })}
      >
        {Logo ? (
          <div
            css={mq({
              display: 'flex',
              flex: 1,
              justifyContent: ['center', 'center', 'flex-start'],
              visibility: ['hidden', 'visible'],
            })}
          >
            <Logo
              css={mq({
                display: ['none', 'initial'],
                margin: [undefined, '0 auto', 0],
              })}
            />
          </div>
        ) : undefined}
        <div css={{ display: 'flex', alignItems: 'center', gap: 8 }}>
          {notificationHandler ? (
            <IconButton onClick={notificationHandler}>
              <Indicator visible={hasNotification}>
                <Notifications color={color.neutral[400]} size={32} />
              </Indicator>
            </IconButton>
          ) : undefined}
          {messageHandler ? (
            <IconButton
              css={mq({ display: ['none', 'inline-block'] })}
              onClick={messageHandler}
            >
              <Indicator visible={hasMessage}>
                <Mail color={color.neutral[400]} size={32} />
              </Indicator>
            </IconButton>
          ) : undefined}
          {avatar && profileHandler ? (
            <IconButton
              css={mq({ display: ['none', 'inline-block'] })}
              onClick={profileHandler}
              variant="rounded"
            >
              <Avatar size={40} src={avatar} />
            </IconButton>
          ) : undefined}
        </div>
      </div>
    </header>
  );
};

export default AppBar;
