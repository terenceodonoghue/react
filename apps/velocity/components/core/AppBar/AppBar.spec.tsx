import { faker } from '@faker-js/faker';
import { render } from '../../utils/test-utils';
import AppBar from './AppBar';

describe('AppBar', () => {
  it('renders content', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(<AppBar>{content}</AppBar>);

    expect(getByTestId('app-bar')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });
});
