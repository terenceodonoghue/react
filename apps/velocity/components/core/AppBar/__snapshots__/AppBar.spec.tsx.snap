// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders content 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border-bottom: solid 1px #e4e8f0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 80px;
  padding: 0 25px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1100;
}

<div>
  <div
    class="emotion-0"
    data-testid="app-bar"
  >
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
  </div>
</div>
`;
